packages/iris-grid/src/IrisGridRenderer.ts

```diff
@@ -1076,6 +1077,53 @@ class IrisGridRenderer extends GridRenderer {
 
     context.restore();
   }
+
+  getExpandButtonPosition(
+    {
+      mouseX,
+      mouseY,
+      metrics,
+      theme,
+    }: {
+      mouseX: Coordinate | null;
+      mouseY: Coordinate | null;
+      metrics: GridMetrics | undefined;
+      theme: GridThemeType;
+    },
+    depth: number | null
+  ): {
+    left: Coordinate | null;
+    top: Coordinate | null;
+    width: number | null;
+    height: number | null;
+  } {
+    const NULL_POSITION = { left: null, top: null, width: null, height: null };
+    if (mouseX == null || mouseY == null || depth == null || !metrics) {
+      return NULL_POSITION;
+    }
+    const { rowHeight, left, top } = GridUtils.getCellInfoFromXY(
+      mouseX,
+      mouseY,
+      metrics
+    );
+
+    assertNotNull(left);
+    assertNotNull(rowHeight);
+    assertNotNull(top);
+    const { cellHorizontalPadding } = theme;
+
+    const width = EXPAND_ICON_SIZE + 2 * cellHorizontalPadding;
+
+    const buttonLeft = left + EXPAND_ICON_SIZE * depth;
```

Comment:
![image](https://user-images.githubusercontent.com/1576283/216644717-e434b01f-0518-4d1b-a942-520a76886324.png)

You'll want to take into account the edge of the panel, so it doesn't ever go offscreen/on to the next panel.